plugins/permission-node/src/types.ts

```diff
@@ -36,23 +37,32 @@ export type PermissionRule<
   TResource,
   TQuery,
   TResourceType extends string,
-  TParams extends unknown[] = unknown[],
+  TParams extends Record<string, unknown> = Record<string, unknown>,
 > = {
   name: string;
   description: string;
   resourceType: TResourceType;
 
+  /**
+   * A ZodSchema that documents the parameters that this rule accepts.
+   */
+  schema: z.ZodObject<{
+    [P in keyof TParams]-?: TParams[P] extends undefined
+      ? z.ZodOptionalType<z.ZodType<TParams[P]>>
```

Comment:
Do you think we should consider starting with a limited set of types that we accept for parameters? It might be simplifying for consumers of this schema to know that (for example), the only parameters that will be included are [`JsonPrimitive`](https://github.com/backstage/backstage/blob/master/packages/types/src/json.ts#L22)s or arrays thereof.

Reply:
This is a great idea. I've made a change to reflect this 👍